modules/imaging_uploader/jsx/UploadForm.js

```diff
@@ -13,27 +13,35 @@ class UploadForm extends React.Component {
   constructor(props) {
     super(props);
 
+    const form = JSON.parse(JSON.stringify(this.props.form));
+    form.IsPhantom.required = true;
+    form.candID.required = true;
+    form.pSCID.required = true;
+    form.visitLabel.required = true;
+    form.mri_file.required = true;
+
     this.state = {
       formData: {},
-      form: JSON.parse(JSON.stringify(this.props.form)),
+      form: form,
       uploadProgress: -1
     };
 
     this.onFormChange = this.onFormChange.bind(this);
+    this.submitForm = this.submitForm.bind(this);
     this.uploadFile = this.uploadFile.bind(this);
   }
 
   componentDidMount() {
-    const form = this.state.form;
-    form.IsPhantom.required = true;
-
     // Disable fields on initial load
-    this.onFormChange(form.IsPhantom.name, null);
+    this.onFormChange(this.state.form.IsPhantom.name, null);
   }
 
   onFormChange(field, value) {
+
+    if (!field) return;
+
     const form = JSON.parse(JSON.stringify(this.state.form));
-    const formData = JSON.parse(JSON.stringify(this.state.formData));
+    const formData = Object.assign({}, this.state.formData);
```

Comment:
@jacobpenny I discovered that deep cloning object using JSON loses file data. Since `formData` does not have child object `Object.assign` seem to do a good enough deep clone here.